src/c1_koiter_steigmann/koiter_steigmann_curvable_bell_elements.h

```diff
@@ -9,580 +9,1183 @@
+	      internal_local_eqn(Index_of_lagrange_data, lagrange_index);
+	    // If this dof isn't pinned we add to the residual
+	    if (lagrange_eqn_number >= 0)
+	    {
+	      // Add contributions from right node
+	      unsigned right_uidadb_index = i_field * n_type + (3 + alpha + beta);
+	      residuals[lagrange_eqn_number] +=
+		right_value[right_uidadb_index];
+	      // Loop over the left node derivatives
+	      for (unsigned gamma = 0; gamma < 2; gamma++)
+	      {
+		// Add contributions from left node first derivatives
+		unsigned left_uidg_index = i_field * n_type + (1 + gamma);
+		residuals[lagrange_eqn_number] +=
+		  - left_value[left_uidg_index]
+		  * hess_of_transform[gamma](alpha, beta);
+		// Loop over the left derivatives again to get second
+		// derivatives
+		for (unsigned delta = 0; delta < 2; delta++)
+		{
+		  // Add contributions from left node second derivatives
+		  unsigned left_uidgdd_index =
+		    i_field * n_type + (3 + gamma + delta);
+		  residuals[lagrange_eqn_number] +=
+		    - left_value[left_uidgdd_index]
+		    * jac_of_transform(gamma, alpha)
+		    * jac_of_transform(delta, beta);
+		}
+	      }
+	      // Jacobian is added during right and left nodal equations using
+	      // symmetry
+	    } // End if eqn not pinned
+          }
+        }
+      } // End loop over displacements [i_field]
+    } // End fill_in_generic_residual_contribution_constraint
 
-    /// Required  # of `values' (pinned or dofs)
-    /// at node n
-    inline unsigned required_nvalue() const
-    {
-      return Initial_Nvalue;
-    }
 
+   
+  }; // End of DuplicateNodeConstraintElement class definition
 
 
+ 
 
-  protected:
+  //===============================================================================
+  /// KoiterSteigmannC1CurvableBellElement elements are a subparametric
+  /// scheme with  linear Lagrange interpolation for approximating the geometry
+  /// and the C1-functions for approximating variables.
+  //==============================================================================
+
+  // [zdec] NNODE_1D(=2) should not be required here due to the fact that all
+  // node should be vertex nodes? Is any templating needed???
+  // template<unsigned DIM,
+  //          unsigned NNODE_1D,
+  //          unsigned BOUNDARY_ORDER,
+  //          template<unsigned DIM_, unsigned NNODE_1D_>
+  //          class PLATE_EQUATIONS>
+  class KoiterSteigmannC1CurvableBellElement
+    : public virtual CurvableBellElement<2>,
+      public virtual KoiterSteigmannEquations
+  {
+  public:
+
+    //----------------------------------------------------------------------
+    // Class construction
+
+    /// Constructor: Call constructors for C1CurvedBellElement and
+    /// KoiterSteigmannEquations
+    KoiterSteigmannC1CurvableBellElement()
+      : CurvableBellElement<Nnode_1D>(Nfield, Field_is_bell_interpolated),
+        KoiterSteigmannEquations()
+    {
+
+     // hierher Aidan: Can this go; is this handled elsewhere?
```

Comment:
As before, this shouldn't have been commented out in the version you got, it was done for debugging. We should talk about integration schemes.